### PR TITLE
Fixes #3866 Update AZEnterpriseAttributesMigrationSync, Migrate Tools

### LINKED ISSUE #3866: Update AZEnterpriseAttributesMigrationSync
## Motivation
`AZEnterpriseAttributesMigrationSync`'s parent class, `MigrationImportSync` (in `migrate_tools`) has been changed in version `6.0.5` to no longer use state for its storage. We need to update our implementation to match. 

This event subscriber is responsible for unpublishing enterprise attributes which disappear from the feed.

### Describe the solution you'd like
`AZEnterpriseAttributesMigrationSync` should be refactored to not use state for storage.

### PR DESCRIPTION
**Pull request by tadean**
## Description
This PR updates to `migrate_tools` version `6.0.5` and implements an API change to `AZEnterpriseAttributesMigrationSync` which inherits from the `migrate_tools` class `MigrationImportSync`

This mimics changes in the parent class to avoid using Drupal state for storage.

This eventsubscriber is responsible for unpublishing attributes which disappear from the attributes feed (rather than the normal behavior, of removing them)

## Related issues
#3866 

## How to test
**BEFORE building:**

- Fetch a copy of the json at `https://api.trellis.arizona.edu/ws/rest/events/v1/enterprise-attributes`
- Place this as `data.json` in the root of your repo folder
- Edit `modules/custom/az_core/az_enterprise_attributes_import/config/install/az_enterprise_attributes_import.settings.yml` to instead have the endpoint `http://localhost/profiles/custom/az_quickstart/data.json` - this will allow you to simulate the endpoint attributes changing.

**After build:**
- enable `az_enterprise_attributes_import`
- verify that Enterprise Attributes have been imported to taxonomy vocabulary
- edit your copy of `data.json` to remove some of terms
- run `drush mim az_enterprise_attributes_import --sync --update`
- Verify you receive a log message saying that the terms you removed are unpublished
- Verify the terms in question are actually unpublished

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [x] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [x] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [x] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] My change requires release notes.

### CODE REVIEW

**Comment by tadean** [comment]
Added a basic unit test. The event subscriber which unpublishes the term doesn't seem to be running during phpunit.

- [ ] Resolve test failure.

**Comment by tadean** [comment]
Resolved test error by making sure the migration actually runs in sync mode.

**Review by bberndt-uaz — DISMISSED** [review]
Looks good! I deleted three terms and they were correctly logged as unpublished:

image

File: modules/custom/az_core/az_enterprise_attributes_import/tests/src/Functional/EnterpriseAttributesImportTest.php [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+use Drupal\Tests\BrowserTestBase;
+use Drupal\migrate\MigrateMessage;
+use Drupal\migrate_tools\MigrateExecutable;
+
+/**
+ * Test of attribute import functionality.
+ *
+ * @ingroup az_enterprise_attributes_import
+ *
+ * @group az_enterprise_attributes_import
```
Comment (bberndt-uaz) [inline modules/custom/az_core/az_enterprise_attributes_import/tests/src/Functional/EnterpriseAttributesImportTest.php:14]: @tadean Just curious: are both of these annotations needed?

Also, I see that Drupal 11 supports tests that use attributes in place of annotations: https://www.drupal.org/node/3447698

Comment (tadean) [reply modules/custom/az_core/az_enterprise_attributes_import/tests/src/Functional/EnterpriseAttributesImportTest.php:14]: I've removed the superfluous group declaration, it doesn't seem to be needed. Thanks @bberndt-uaz !

I also tried out the attribute version, but it does look like it's restricted to PHPunit 10 in D11.

**Review by trackleft — DISMISSED** [review]

**Review by joeparsons — APPROVED** [review]

**Review by trackleft — APPROVED** [review]

**Pull request merged**